pkg/deployments/tasks/20230217-gnosis-root-gauge-factory/test/task.fork.ts

```diff
@@ -0,0 +1,326 @@
+    // First grant permission to call add_type to the admin.
+    await authorizer
+      .connect(govMultisig)
+      .grantRole(
+        await authorizerAdaptor.getActionId(gaugeController.interface.getSighash('add_type(string,uint256)')),
+        admin.address
+      );
+
+    await authorizer
+      .connect(govMultisig)
+      .grantRole(
+        await authorizerAdaptor.getActionId(gaugeController.interface.getSighash('add_gauge(address,int128)')),
+        admin.address
+      );
+  });
+
+  it('add gauge factory', async () => {
+    // Not actually using this, but leaving it in to document the process
+
+    // Because the Vyper `add_type` function has a default argument, we need to access it using the full signature, in the Sighash for the
+    // permission (above), and when actually calling it with arguments (below).
+    await authorizerAdaptor
+      .connect(admin)
+      .performAction(
+        gaugeController.address,
+        gaugeController.interface.encodeFunctionData('add_type(string,uint256)', ['Gnosis', 1])
+      );
+
+    await gaugeAdder.addGaugeFactory(factory.address, GaugeType.Gnosis);
+  });
+
+  it('add gauge to gauge controller', async () => {
+    // Add gauge directly through the controller, since we can't use GaugeAdder V3 without the TimelockAuthorizer migration
+    //await gaugeAdder.addGnosisGauge(gauge.address);
+
+    await authorizerAdaptor
+      .connect(admin)
+      .performAction(
+        gaugeController.address,
+        gaugeController.interface.encodeFunctionData('add_gauge(address,int128)', [gauge.address, GaugeType.Gnosis])
+      );
+
+    expect(await gaugeController.gauge_exists(gauge.address)).to.be.true;
+  });
+
+  it('vote for gauge', async () => {
+    expect(await gaugeController.get_gauge_weight(gauge.address)).to.equal(0);
+    expect(await gauge.getCappedRelativeWeight(await currentTimestamp())).to.equal(0);
+
+    await gaugeController.connect(veBALHolder).vote_for_gauge_weights(gauge.address, 10000); // Max voting power is 10k points
+
+    // We now need to go through an epoch for the votes to be locked in
+    await advanceTime(DAY * 8);
+
+    await gaugeController.checkpoint();
+
+    // Gauge weight is equal to the cap, and controller weight for the gauge is greater than the cap.
+    expect(
+      await gaugeController['gauge_relative_weight(address,uint256)'](gauge.address, await currentWeekTimestamp())
+    ).to.be.gt(weightCap);
+    expect(await gauge.getCappedRelativeWeight(await currentTimestamp())).to.equal(weightCap);
+  });
+
+  it('mint & bridge tokens', async () => {
+    // The gauge has votes for this week, and it will mint the first batch of tokens. We store the current gauge
+    // relative weight, as it will change as time goes by due to vote decay.
+    const firstMintWeekTimestamp = await currentWeekTimestamp();
+    //const gaugeRelativeWeight = await gaugeController['gauge_relative_weight(address)'](gauge.address);
```

Comment:
Why is this commented?

Reply:
I believe it was leftover from Tom's old PR, which I had to update looking at more recent ones.